### PR TITLE
Rhodes documentation map

### PR DESCRIPTION
**Pull request by david-rhodes**
Source code documentation pass for `Map` pieces. See latest published documents here: https://mapbox.github.io/mapbox-sdk-cs/api/Mapbox.Map.html.

I also removed the Update side effects from `Map.cs` and updated tests.

### CODE REVIEW

File: src/Map/TileCover.cs [hunk 1]
```diff
@@ -49,7 +76,15 @@ public static class TileCover {
 		/// <param name="coord"> Geographic coordinate. </param>
 		/// <param name="zoom"> Zoom level. </param>
 		/// <returns>The to tile identifier.</returns>
-		public static UnwrappedTileId CoordinateToTileId(Vector2d coord, int zoom) {
+		/// <example>
+		/// Convert a geocoordinate to a TileId:
+		/// <code>
+		/// var unwrappedTileId = TileCover.CoordinateToTileId(new Vector2d(40.015, -105.2705), 18);
+		/// Debug.Log("UnwrappedTileId: " + unwrappedTileId.ToString());
```
Comment (isiyu) [inline src/Map/TileCover.cs]: `Debug.Log` is Unity specific right? 
Should we make this not depend on unity. `Console.write` should work here in both Unity and Non-unity cases?

File: src/Map/VectorTile.cs [hunk 2]
```diff
@@ -4,25 +4,44 @@
...
+	/// 
+	/// // Make the request.
+	/// vectorTile.Initialize(parameters, (Action)(() =>
+	/// {
+	/// 	if (!string.IsNullOrEmpty(vectorTile.Error))
+	/// 	{
+	///			// Handle the error.
+	///		}
+	/// 
+	/// 	// Consume the <see cref="Data"/>.
```
Comment (isiyu) [inline src/Map/VectorTile.cs]: docfx formatting issue:
screen shot 2017-04-11 at 4 11 00 pm

File: src/Map/RasterTile.cs [hunk 3]
```diff
@@ -4,40 +4,66 @@
...
+	/// 
+	/// // Make the request.
+	/// rasterTile.Initialize(parameters, (Action)(() =>
+	/// {
+	/// 	if (!string.IsNullOrEmpty(rasterTile.Error))
+	/// 	{
+	///			// Handle the error.
+	///		}
+	/// 
+	/// 	// Consume the <see cref="Data"/>.
```
Comment (BergWerkGIS) [inline src/Map/RasterTile.cs]: We saw `xref` problems before.
Updating `docfx` solved it.

In root of `mapbox-unity-sdk` delete
* file  `docfx.zip`
* folder `docfx`

and then run `./scripts/build-docs.sh` again.

Does it work now?

File: src/Map/Map.cs [hunk 4]
```diff
@@ -6,20 +6,35 @@
...
+	using System.Collections.Generic;
+	using Mapbox.Platform;
+	using Mapbox.Utils;
+
+	/// <summary>
+	///     The Mapbox Map abstraction will take care of fetching and decoding
+	///     data for a geographic bounding box at a certain zoom level.
+	/// </summary>
+	/// <typeparam name="T">
+	///     The tile type, currently <see cref="T:Mapbox.Map.Vector"/> or
```
Comment (david-rhodes) [inline src/Map/Map.cs]: You're looking at the wrong repo. Use: https://mapbox.github.io/mapbox-unity-sdk/api/mapbox-sdk-cs/Mapbox.Map.Map-1.html or https://mapbox.github.io/mapbox-sdk-cs/api/Mapbox.Map.Map-1.html.

**Pull request merged**